src/comestible_inv.cpp

```diff
@@ -0,0 +1,780 @@
+        {comestible_inv_area::get_info( comestible_inv_area_info::AIM_ALL_I_W )}
+    }
+} )
+{
+    //TODO:XXX
+    // initialize screen coordinates for small overview 3x3 grid, depending on control scheme
+    //if (tile_iso && use_tiles) {
+    //  // Rotate the coordinates.
+    //  squares[1].info.hscreen.x = 33;
+    //  squares[2].info.hscreen.x = 36;
+    //  squares[3].info.hscreen.y = 2;
+    //  squares[4].info.hscreen.y = 3;
+    //  squares[6].info.hscreen.y = 1;
+    //  squares[7].info.hscreen.y = 2;
+    //  squares[8].info.hscreen.x = 30;
+    //  squares[9].info.hscreen.x = 33;
+    //}
+}
+
+comestible_inventory::~comestible_inventory()
+{
+    save_settings();
+    auto &aim_code = uistate.comestible_save.exit_code;
+    if( aim_code != exit_re_entry ) {
+        aim_code = exit_okay;
+    }
+    // Only refresh if we exited manually, otherwise we're going to be right back
+    if( exit ) {
+        werase( head );
+        werase( minimap );
+        werase( mm_border );
+        werase( window );
+        g->refresh_all();
+        g->u.check_item_encumbrance_flag();
+    }
+    delete pane;
+}
+
+void comestible_inventory::save_settings()
+{
+    bool is_reset = uistate.comestible_save.exit_code != exit_re_entry;
+    pane->save_settings( is_reset );
+    if( is_reset ) {
+        if( uistate.comestible_save.bio != -1 ) {
+            g->u.deactivate_bionic( uistate.comestible_save.bio );
+        }
+        uistate.comestible_save.bio = -1;
+    }
+}
+
+comestible_inv_area *comestible_inventory::get_area( const std::string &action )
+{
+    for( auto &i : squares ) {
+        if( action == i.info.actionname ) {
+            return &i;
+        }
+    }
+    return nullptr;
+}
+
+void comestible_inventory::init()
+{
+    for( auto &square : squares ) {
+        square.init();
+    }
+
+    w_height = TERMY < min_w_height + head_height ? min_w_height : TERMY - head_height;
+    w_width = TERMX < min_w_width ? min_w_width : TERMX > max_w_width ? max_w_width :
+              static_cast<int>( TERMX );
+
+    headstart = 0;
+    colstart = TERMX > w_width ? ( TERMX - w_width ) / 2 : 0;
+
+    head = catacurses::newwin( head_height, w_width - minimap_width, point( colstart, headstart ) );
+    mm_border = catacurses::newwin( minimap_height + 2, minimap_width + 2,
+                                    point( colstart + ( w_width - ( minimap_width + 2 ) ), headstart ) );
+    minimap = catacurses::newwin( minimap_height, minimap_width,
+                                  point( colstart + ( w_width - ( minimap_width + 1 ) ), headstart + 1 ) );
+    window = catacurses::newwin( w_height, w_width, point( colstart, headstart + head_height ) );
+
+    uistate.comestible_save.exit_code = exit_none;
+
+    //pane is set by child
+    int itemsPerPage = w_height - 2 - 5; // 2 for the borders, 5 for the header stuff
+    pane->init( itemsPerPage, window, &squares );
```

Comment:
This is the only place that actually calls `init` (all other places are within the various `init` functions). It's the only place that provides a value for the `a` parameter, and that value is never null. It should be a reference.

Reply:
I changed pointers to references in a few places. I end up with code like
```
Class C{};

C* var;
void set(int& v){var = &v;}
C& get(){return *C;}
```
Feels kind of weird to pass reference, but store as a pointer. Is that the right way to do it?

Reply:
Now. Do the initialization in the constructor, that is what it is there for.

```C++
Class C{
 std::array<...> &all_squares;
 C( std::array<...> &a ) : all_squares( a ) { }
};
```
Don't write a setter function that you don't need. `all_areas` is never going to change after it has been set once, so you can set it directly via the constructor and that's it.